### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by Joel-100**
- Created app.rb file that will serve as your console app entry point.
- Created main.rb which defines the entry point
- Created separated files create_student, create_teacher, rent_out_book.rb and list_rentals.rb

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -29,4 +29,8 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
 end
```
Comment (tresorsawasawa) [inline person.rb:37]: - [ ] Kindly make sure you correct the typing error while importing your files at the top of this file(`person.rb`) as it raises an error while running your application. 
```ruby
# Bad
require './captalize'

# Good
require './capitalize'
```

File: rent_out_book.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      end
+    end
+    person_id = gets.chomp.to_i
+    print 'Date (yyyy/mm/d): '
+    date_of_rental = gets.chomp
+    @rentals.push(Rental.new(date_of_rental, @persons[person_id], @books[book_id]))
+    puts 'Rental created successfully'
+    puts ''
+    library_menu
+  end
```
Comment (tresorsawasawa) [inline rent_out_book.rb:38]: - [ ] Kindly make sure the user is able to `Create a rental` as shown in the template **[video](https://www.youtube.com/watch?v=vkkgrhD6aXQ)**. Right now, when clicking the **Option 5 for creating a rental**, after the user has entered the `date`, your app shows the below error, make sure you fix it:
Screenshot from 2022-12-22 21-57-31

File: list_rentals.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      puts person_id
+      if rental.person.id.to_i == person_id
+        puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}"
+      else
+        puts 'Sorry, no rentals found!'
+        puts ''
+        puts 'Enter 5 to create a rental for the person'
+        select_option
+      end
+    end
```
Comment (tresorsawasawa) [inline list_rentals.rb:37]: - [ ] As option **5** for creating a rental is not working, it is impossible to also check if the **6th option of listing all rentals for a given person is** working as expected. By this I mean, once you have fixed the issue of creating a rental, you should make sure the user is able to list **all rentals for a given person** to make your App be complete. 

Good :+1: job as other options are working as expected

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Joel-100 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @Joel-100,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**